server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/branches/ProjectBranchesApi.java

```diff
@@ -37,31 +38,31 @@ public ProjectBranchesApi(ServerApiHelper helper) {
     this.helper = helper;
   }
 
-  public Collection<String> getAllBranchNames(String projectKey) {
+  public Set<ServerBranch> getAllBranches(String projectKey) {
     var response = helper.get(LIST_ALL_PROJECT_BRANCHES_URL + "?project=" + projectKey);
     var bodyAsString = response.bodyAsString();
     return getBranchNamesFromResponse(bodyAsString);
   }
 
-  private static Collection<String> getBranchNamesFromResponse(String bodyAsString) {
-    Collection<String> parsedBranchNames = new ArrayList<>();
+  static Set<ServerBranch> getBranchNamesFromResponse(String bodyAsString) {
+    Set<ServerBranch> parsedBranchNames = new HashSet<>();
     try {
       var root = JsonParser.parseString(bodyAsString).getAsJsonObject();
       var branches = root.get("branches").getAsJsonArray();
 
       for (JsonElement el : branches) {
         var branch = el.getAsJsonObject();
-        var element = branch.get("name");
-        if (element == null) {
+        var name = branch.get("name");
+        var isMain = branch.get("isMain");
+        if (name == null || isMain == null) {
```

Comment:
The error message is not consistent with the conditions tested here: if `isMain` is `null`, we'll get a message that mentions `name` 🤔